webpack/assets/javascripts/react_app/components/common/Slot/SlotSelectors.js

```diff
@@ -36,3 +39,15 @@ export const selectFillsComponents = (state, props) => {
   }
   return [];
 };
+
+export const selectSlotMetadata = (state, id) => {
+  const registerdFills = state.extendable[id];
+  let slotMetadata = {};
+  // eslint bug - https://github.com/eslint/eslint/issues/12117
+  /* eslint-disable-next-line no-unused-vars */
+  for (const fill in registerdFills) {
+    if (registerdFills[fill].metadata)
+      slotMetadata = merge(slotMetadata, registerdFills[fill].metadata);
+  }
+  return slotMetadata;
+};
```

Comment:
How about something like:
```suggestion
export const selectSlotMetadata = (state, id) => {
  const registerdFills = state.extendable[id];
  const slotMetadata = {};
  // eslint bug - https://github.com/eslint/eslint/issues/12117
  /* eslint-disable-next-line no-unused-vars */
  for (const fill in registerdFills) {
    if (registerdFills[fill].metadata)
      slotMetadata[fill] = registerdFills[fill].metadata;
  }
  return slotMetadata;
};
```
and then each fill's metadata is accessible using the fill id and there is no risk of one fill overriding another fill's metadata?

TBH the whole data structure for how we store fills should probably be changed so we don't have the division and duplication of attributes between the registry and the store, but that's outside the scope for this PR.